### ISSUE TITLE
Read Distribution

### ISSUE THREAD

**Issue by j-berg** [XPRESSyourself/XPRESSpipe#23]
```
Performing read distribution analysis on fastq files...
Resetting parallelization specs based on max file size to be processed:
Max number of workers: 5
Number of threads per worker (where available): 6
trimmed_SRR1795409_1.fastq has been processed.
trimmed_SRR1795410_1.fastq has been processed.
trimmed_SRR1795411_1.fastq has been processed.
trimmed_SRR1795412_1.fastq has been processed.
trimmed_SRR1795413_1.fastq has been processed.
Processing of batch 1 of 7 complete...
trimmed_SRR1795414_1.fastq has been processed.
trimmed_SRR1795415_1.fastq has been processed.
trimmed_SRR1795416_1.fastq has been processed.
trimmed_SRR1795417_1.fastq has been processed.
trimmed_SRR1795418_1.fastq has been processed.
Processing of batch 2 of 7 complete...
trimmed_SRR1795419_1.fastq has been processed.
trimmed_SRR1795420_1.fastq has been processed.
trimmed_SRR1795421_1.fastq has been processed.
trimmed_SRR1795422_1.fastq has been processed.
trimmed_SRR1795423_1.fastq has been processed.
Processing of batch 3 of 7 complete...
Traceback (most recent call last):
  File "/uufs/chpc.utah.edu/common/home/u0690617/miniconda3/bin/xpresspipe", line 11, in <module>
    load_entry_point('XPRESSpipe', 'console_scripts', 'xpresspipe')()
  File "/uufs/chpc.utah.edu/common/home/u0690617/XPRESSpipe/xpresspipe/__main__.py", line 309, in main
    make_readDistributions(args_dict)
  File "/uufs/chpc.utah.edu/common/home/u0690617/XPRESSpipe/xpresspipe/readDistribution.py", line 127, in make_readDistributions
    mod_workers = True)
  File "/uufs/chpc.utah.edu/common/home/u0690617/XPRESSpipe/xpresspipe/parallel.py", line 158, in parallelize
    args_dict)
  File "/uufs/chpc.utah.edu/common/home/u0690617/XPRESSpipe/xpresspipe/parallel.py", line 124, in run_pools
    for file in zip(batch, executor.map(func, batch)):
  File "/uufs/chpc.utah.edu/common/home/u0690617/miniconda3/lib/python3.7/concurrent/futures/process.py", line 476, in _chain_from_iterable_of_lists
    for element in iterable:
  File "/uufs/chpc.utah.edu/common/home/u0690617/miniconda3/lib/python3.7/concurrent/futures/_base.py", line 586, in result_iterator
    yield fs.pop().result()
  File "/uufs/chpc.utah.edu/common/home/u0690617/miniconda3/lib/python3.7/concurrent/futures/_base.py", line 432, in result
    return self.__get_result()
  File "/uufs/chpc.utah.edu/common/home/u0690617/miniconda3/lib/python3.7/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
concurrent.futures.process.BrokenProcessPool: A process in the process pool was terminated abruptly while the future was running or pending.
```

**Comment by j-berg** [comment]
Appears to be an issue where it correctly processes the 1-3 batches, and then dies. Not dumping memory?

**Comment by j-berg** [comment]
Fixed memory issues with commit 85939d84f8810d3612603caa5ce094661e592dde by converting process to Julia script. No longer requires memory threshold.
